src/components/transactions/FilterModal.tsx

```diff
@@ -0,0 +1,393 @@
+import { TransactionStatus } from '../../types/types';
+import FilterTag from './FilterTag';
+
+type FilterModalProps = {
+  filterState: FilterState;
+  filterDispatch: FilterDispatch;
+  sortDispatch: SortTransactionDispatch;
+  isVisible: boolean;
+  setIsVisible: (visibility: boolean) => void;
+  minDatePickerIsVisible: boolean;
+  setMinDatePickerIsVisible: (visibility: boolean) => void;
+  maxDatePickerIsVisible: boolean;
+  setMaxDatePickerIsVisible: (visibility: boolean) => void;
+};
+
+export default function FilterModal({
+  filterState,
+  filterDispatch,
+  sortDispatch,
+  isVisible,
+  setIsVisible,
+  minDatePickerIsVisible,
+  setMinDatePickerIsVisible,
+  maxDatePickerIsVisible,
+  setMaxDatePickerIsVisible,
+}: FilterModalProps) {
+  return (
+    <Modal
+      isVisible={isVisible}
+      coverScreen={false}
+      style={Styles.modal}
+      backdropTransitionOutTiming={0}
+    >
+      <FilterModalContainer>
+        <CloseButtonContainer>
+          <TouchableOpacity
+            onPress={() => {
+              setIsVisible(false);
+              filterDispatch({ type: 'RESET_IN_PROGRESS' });
+            }}
+          >
+            <BlueText>
+              <Body1Text>Close</Body1Text>
+            </BlueText>
+          </TouchableOpacity>
+        </CloseButtonContainer>
+
+        <PaddedScrollView alwaysBounceVertical={false}>
+          <CenteredTextContainer>
+            <H4CardNavTab>Filter Invoices</H4CardNavTab>
+          </CenteredTextContainer>
+
+          <SubheadingContainer>
+            <OneLine>
+              <Body1SemiboldText>Filter by date</Body1SemiboldText>
+              <ClearButton
+                isDisabled={
+                  !(
+                    filterState.inProgressMinDateIsSet ||
+                    filterState.inProgressMaxDateIsSet
+                  )
+                }
+                onPress={() => {
+                  filterDispatch({
+                    type: 'CLEAR_DATE_FILTERS',
+                  });
+                }}
+              />
+            </OneLine>
+          </SubheadingContainer>
+
+          <OneLine style={{ zIndex: 5 }}>
+            <LeftAlignContainer>
+              <FilterField
+                isSelected={filterState.inProgressMinDateIsSet}
+                onPress={() => {
+                  if (maxDatePickerIsVisible) {
+                    setMaxDatePickerIsVisible(false);
+                  }
+                  setMinDatePickerIsVisible(true);
+                }}
+                icon="calendar-today"
+              >
+                <Body1Text>
+                  {filterState.inProgressMinDateIsSet ? (
+                    moment(filterState.inProgressMinDate).format('M/DD/YYYY')
+                  ) : (
+                    <MidGrayText>From</MidGrayText>
+                  )}
+                </Body1Text>
+              </FilterField>
+            </LeftAlignContainer>
+            <HorizontalSpacing />
```

Comment:
Can we create a filter-by-date container that positions the date boxes? It'd clean up our styling by removing styled components used only for spacing.